### PR TITLE
Setup Kafka, MongoDb, Deployment Pipe & Sonar

### PR DESCRIPTION
**Pull request by SahilSinghStudierender**

### CODE REVIEW

File: src/main/java/at/ac/fhcampuswien/se_booking/model/kafka/KafkaMessageBaseModel.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package at.ac.fhcampuswien.se_booking.model.kafka;
+
+import lombok.Data;
+
+@Data
+public abstract class KafkaMessageBaseModel {
```
Comment (veidl) [inline src/main/java/at/ac/fhcampuswien/se_booking/model/kafka/KafkaMessageBaseModel.java]: why did you create a base model if its not even used?

File: src/main/java/at/ac/fhcampuswien/se_booking/model/kafka/BookingMessage.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.experimental.FieldDefaults;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class BookingMessage extends KafkaMessageBaseModel{
```
Comment (veidl) [inline src/main/java/at/ac/fhcampuswien/se_booking/model/kafka/BookingMessage.java]: Unsued classes should not be pushed.

**Review by veidl — CHANGES_REQUESTED** [review]

File: .github/workflows/deployment_workflow.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  IBM_CLOUD_API_KEY: ${{ secrets.IBM_CLOUD_API_KEY }}
+  IBM_CLOUD_REGION: eu-de
+  K8S_NAMESPACE: part2 # ADAPT!
+  IMAGE_NAME: se-booking-service-part2 # ADAPT!
+  IKS_CLUSTER: ${{ secrets.IKS_CLUSTER }} # name or id of cluster
+  DEPLOYMENT_NAME: se-booking-service-part2 # ADAPT!
+  INTERNAL_PORT: 8080 # ADAPT!
+  EXTERNAL_PORT: 32769 # ADAPT!
```
Comment (Henchway) [inline .github/workflows/deployment_workflow.yml]: The default node port range for Kubernetes is 30000 - 32767

File: .github/workflows/deployment_workflow.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  DOCKER_USER: ${{ secrets.DOCKER_USER }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  CURRENCY_CONVERTER_URL: http://169.51.206.49:32767/?wsdl
```
Comment (Henchway) [inline .github/workflows/deployment_workflow.yml]: Remove / Adapt to gRPC service port

File: pom.xml [hunk 5]
```diff
@@ -1,54 +1,177 @@
...
+                            <formats>
+                                <format>XML</format>
+                                <format>HTML</format>
+                            </formats>
+                        </configuration>
+                    </execution>
+                </executions>
+                <configuration>
+                    <excludes>
+                        <exclude>at/ac/fhcampuswien/car_rental/soap/**/*</exclude>
```
Comment (veidl) [inline pom.xml]: fix the exclusions

File: .github/workflows/deployment_workflow.yml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      # Deploy the Docker image to the IKS cluster
+      - name: Deploy to IKS
+        # Change the service type to the actual service
+        run: |
+          ibmcloud login --apikey "${IBM_CLOUD_API_KEY}" -r "${IBM_CLOUD_REGION}" -g Default
+          ibmcloud ks cluster config --cluster $IKS_CLUSTER
+          kubectl create namespace $K8S_NAMESPACE --dry-run=client -o yaml > namespace.yaml
+          kubectl apply -f namespace.yaml
+          kubectl config set-context --current --namespace=$K8S_NAMESPACE
+          kubectl create configmap $DEPLOYMENT_NAME --from-literal=CURRENCY_CONVERTER_URL=${CURRENCY_CONVERTER_URL} --from-literal=POSTGRES_SERVICE_NAME=${POSTGRES_SERVICE_NAME} --dry-run -o yaml > configmap.yaml
```
Comment (Henchway) [inline .github/workflows/deployment_workflow.yml]: Config map needs to be adapted probably

**Review by Henchway — CHANGES_REQUESTED** [review]

File: .github/workflows/deployment_workflow.yml (hunk 4 above)
Comment (Henchway) [reply .github/workflows/deployment_workflow.yml]: The gRPC port is 32760

File: .github/workflows/deployment_workflow.yml (hunk 3 above)
Comment (Henchway) [reply .github/workflows/deployment_workflow.yml]: Also: No Dockerfile? This will fail in any case then, also why branch master, not main? :(

**Review by veidl — APPROVED** [review]
lgtm

File: .github/workflows/deployment_workflow.yml (hunk 3 above)
Comment (SahilSinghStudierender) [reply .github/workflows/deployment_workflow.yml]: used 30069

File: .github/workflows/deployment_workflow.yml (hunk 4 above)
Comment (SahilSinghStudierender) [reply .github/workflows/deployment_workflow.yml]: Removed it, will add it when needed

File: .github/workflows/deployment_workflow.yml (hunk 6 above)
Comment (SahilSinghStudierender) [reply .github/workflows/deployment_workflow.yml]: Removed it

**Review by Henchway — APPROVED** [review]

**Pull request merged**